reposerver/repository/repository.go

```diff
@@ -1996,6 +1998,22 @@ func runConfigManagementPluginSidecars(ctx context.Context, appPath, repoPath, p
 		log.Debugf("common root path calculated for application %s: %s", q.AppName, rootPath)
 	}
 
+	pluginConfigResponse, err := cmpClient.CheckPluginConfiguration(ctx, &emptypb.Empty{})
+	if err != nil {
+		return nil, fmt.Errorf("error calling cmp-server checkPluginConfiguration: %w", err)
+	}
```

Comment:
fwiw, this call is made for every plugin invocation earlier in the code, so this change will duplicate the call.

It's a lightweight call, so I'm not worried. Just good to keep in mind.

Reply:
Hmm, tracing the call stack here they are fairly separated in their invocation here. I don't see a way to unify the call to get the plugin config after a few minutes looking through

Reply:
Yeah, you'd have to return this boolean from the `discovery.AppType` call which is weird and won't read well. I'm refactoring a lot of that code anyway, maybe I can make it nicer.

Reply:
does this refactoring makes sense in the context of this PR or in a follow on?

Reply:
Probably a follow-on, it's pretty invasive.